core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java

```diff
@@ -566,9 +566,16 @@ private void openBlock(BlockReadRequestContext context)
           || (reader instanceof DelegatingBlockReader
           && ((DelegatingBlockReader) reader).getDelegate()
           instanceof UnderFileSystemBlockReader)) {
-        AlluxioURI ufsMountPointUri =
-            ((UnderFileSystemBlockReader) ((DelegatingBlockReader) reader).getDelegate())
-                .getUfsMountPointUri();
+        AlluxioURI ufsMountPointUri;
+        if (reader instanceof DelegatingBlockReader) {
```

Comment:
can we do it in two blocks (one for UfsBlockReader, one for DelegatingBlockReader), instead of putting them nested?